From f22e5b0b23fcd59fc965c827b4036b212a603536 Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Thu, 22 Aug 2024 11:20:17 -0400
Subject: [PATCH] Avoid criticals on headless systems

If we don't have a monitor, don't try to use it.
This avoids pointless critical warnings on headless systems.
---
 gtk/gtkwidget.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/gtk/gtkwidget.c b/gtk/gtkwidget.c
index b6e00115bb..23401a44e8 100644
--- a/gtk/gtkwidget.c
+++ b/gtk/gtkwidget.c
@@ -10984,7 +10984,10 @@ gtk_widget_get_scale_factor (GtkWidget *widget)
   display = gtk_widget_get_display (widget);
   monitor = gdk_display_get_monitor (display, 0);
 
-  return gdk_monitor_get_scale_factor (monitor);
+  if (monitor)
+    return gdk_monitor_get_scale_factor (monitor);
+
+  return 1;
 }
 
 /**
-- 
2.48.1

